pkg/internal/goexec/instructions.go

```diff
@@ -88,18 +87,12 @@ func findGoSymbolTable(elfF *elf.File) (*gosym.Table, error) {
 			return nil, fmt.Errorf("acquiring .gopclntab data: %w", err)
 		}
 	}
-	sec := elfF.Section(".gosymtab")
-	if sec == nil {
-		return nil, errors.New(".gosymtab section not found in target binary, make sure this is a Go application")
-	}
-	symTabRaw, err := sec.Data()
-	if err != nil {
-		return nil, fmt.Errorf("acquiring .gosymtab data: %w", err)
-	}
 	pcln := gosym.NewLineTable(pclndat, elfF.Section(".text").Addr)
-	symTab, err := gosym.NewTable(symTabRaw, pcln)
+	// First argument accepts the .gosymtab ELF section.
+	// Since Go 1.3, .gosymtab is empty so we just pass an nil slice
+	symTab, err := gosym.NewTable(nil, pcln)
```

Comment:
Since the generated executable does not contain `.gosymtab` section, the load failed. However, I verified that the `.gosymtab` contents are an empty slice and passing a nil slice does not have any effect (but Beyla doesn't fail if it doesn't find it)